src/main/webapp/app/shared/dashboards/tutor-leaderboard/tutor-leaderboard.component.html

```diff
@@ -55,7 +55,7 @@
                 </td>
                 <td>{{ tutor.points }}</td>
                 <td>{{ tutor.numberOfAssessments }}</td>
-                <ng-container *ngIf="course && !course.isAtLeastInstructor; else atLeastInstructor">
+                <ng-container *ngIf="course && !course.isAtLeastInstructor && (!exercise || !exercise.isAtLeastInstructor); else atLeastInstructor">
```

Comment:
When we are in the assessment dashboard of an exam exercise, the course (i.e. exercise.exam.course) doesn't have the isAtLeastInstructor property. Instead it's a property of the exercise itself. So we check if the user is either an instructor of the course or of the exercise.

Reply:
This does not seem like the right solution to me here. If the property is not set, the service layer probably does not handle the result correctly, and we should fix that.

Reply:
I think the value in the exercise should always be present regardless if it is a course exercise or an exam exercise, so this should work too:
```suggestion
                <ng-container *ngIf="!exercise || !exercise.isAtLeastInstructor; else atLeastInstructor">
```

Reply:
But this component is also used for the exam dashboard where exercise is not set (the leaderboard there includes all exercises of the exam). So if we are in the exam dashboard (meaning exercise is not set) we need to check for course instructor, otherwise for exercise instructor.

Reply:
And for exam exercises the course-property is unset? That's what I meant by "the service layer probably does not handle the result correctly". Here, you should be able to expect `course.isAtLeastInstructor` to be set, and if not this should get changed.

Reply:
You're right, the way the course was accessed in the account service was wrong